### ISSUE TITLE
TsLint: assignment operation inside an 'if' condition should be marked with a warning

### ISSUE THREAD

**Issue by AGiorgetti** [Microsoft/TypeScript#3890]
If we want to spot some more bugs at compile time, I believe that something like:

```javascript
if (data.id = id) { ...
```

while being perfectly legal, it can be a coding bug most of the times.
it would be nice if TsLint could spot this type of code and give us a warning instead of using a custom regex that scan all our files.

**Comment by DickvdBrink** [comment]
TSLint issues should be reported here: https://github.com/palantir/tslint/ as that isn't a Microsoft thing :)
Definitely nice to have! :)

**Comment by AGiorgetti** [comment]
ops sorry, will open the ticket there, thanks!
